pyproject.toml

```diff
@@ -41,15 +41,14 @@ Changelog = "https://pypdf.readthedocs.io/en/latest/meta/CHANGELOG.html"
 
 [project.optional-dependencies]
 full = [
-    "cryptography; python_version >= '3.7'",
-    "PyCryptodome; python_version == '3.6'",
-    "Pillow>=8.0.0",
+    "cryptography; python_version >= '3.8'",
+    "Pillow>=10.3.0",
 ]
 crypto = [
-    "cryptography; python_version >= '3.7'",
-    "PyCryptodome; python_version == '3.6'",
+    "cryptography; python_version >= '3.8'",
+    "PyCryptodome; python_version >= '3.6'",
 ]
-image = ["Pillow>=8.0.0"]
+image = ["Pillow>=10.3.0"]
```

Comment:
There is no need to update this pin.

Reply:
There is an issue on previous version

Reply:
This is wrong if I remember correctly - there is one specific version which does not work. If we would like to avoid users installing it, we could declare it as a condition, but this condition is wrong in my opinion, especially introducing this in the release commit without further discussions.

Reply:
If there is a new feature in 10.0 which we require then `>=10.0` is the way to go.

If we know that it works from 10.0 upwards, but e.g. 10.2 was broken, we can also do something like `>=10.0,!=10.2`. See https://peps.python.org/pep-0440/#examples

> I would not like to allow issues to be raised because not using the good version.

:+1: 

> as we move to drop python 3.7, pillow needs to be on version <=10.0

I guess you mean >= 10.0?

However, in that case Python will automatically select the right one. We should typically only restrict the version if there are features missing (or an API changed so we know it's incompatible)

> why you want to stay on obsolete version ?

It's not that. A library should be as permissive as possible with the version as the application using it will likely also use other libraries. The more restrictive we are, the harder it will be to use pypdf in combination with other libraries.

Users can still install a newer version.